internal/tlsprofile/get.go

```diff
@@ -0,0 +1,38 @@
+package tlsprofile
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	openshiftconfigv1 "github.com/openshift/api/config/v1"
+
+	configv1alpha1 "github.com/os-observability/tempo-operator/apis/config/v1alpha1"
+)
+
+// Get the profile according to the features configuration, if the policy is invalid or is not specified (empty string) this
+// should return the default profile (Intermediate), if openshift.ClusterTLSPolicy is enabled, it should get the profile
+// from the cluster, if the cluster return a unknow profile this should return an error.
+func Get(ctx context.Context, fg configv1alpha1.FeatureGates, c k8getter, log logr.Logger) (TLSProfileOptions, error) {
+	var tlsProfileType openshiftconfigv1.TLSSecurityProfile
+	var err error
+
+	// If ClusterTLSPolicy is enabled get the policy from the cluster
+	if fg.OpenShift.ClusterTLSPolicy {
+		tlsProfileType, err = getTLSProfileFromCluster(ctx, c)
+	} else {
+		tlsProfileType, err = getTLSSecurityProfile(configv1alpha1.TLSProfileType(fg.TLSProfile))
+	}
+
+	if err != nil {
+		log.Error(err, "failed to get security profile. will use default tls profile.")
+		tlsProfileType = getDefaultTLSSecurityProfile()
```

Comment:
I think for security configurations I would put tempo in a degraded/failed state instead of silently falling back to the defaults (I know there's a log, but I assume only very few people actually read the container logs).

Reply:
+1

Reply:
I propose this:

- if the profile is not specified on the gateway flags, we use the default, no degrade state.
- if the `openshift.ClusterTLSPolicy ` is enabled and failed to fetch the policy from the cluster, then we can think on return a degraded error. 

WDYT?

Reply:
> I propose this:
> 
> * if the profile is not specified on the gateway flags, we use the default, no degrade state.
> * if the `openshift.ClusterTLSPolicy ` is enabled and failed to fetch the policy from the cluster, then we can think on return a degraded error.
> 
> WDYT?

I'm not sure about the "failing to fetch the policy from the cluster". I don't have a strong opinion but I think it can be a little bit difficult to detect.

Reply:
> I propose this:
> 
>     * if the profile is not specified on the gateway flags, we use the default, no degrade state.
+1
>     * if the `openshift.ClusterTLSPolicy ` is enabled and failed to fetch the policy from the cluster, then we can think on return a degraded error.
Sounds good!

Reply:
Done with the behavior we agreed.